app/v3/retrieveBalanceAndTransactions/def1/model/response/FinancialDetailsItem.scala

```diff
@@ -52,7 +52,13 @@ object FinancialDetailsItem {
       (__ \ "clearingReason").readNullable[String] and
       (__ \ "outgoingPaymentMethod").readNullable[String].map(paymentMethodConverter) and
       (if (readLocks.value) __.read[FinancialDetailsItemLocks].map(Some(_)) else Reads.pure(None)) and
-      (__ \ "returnFlag").readNullable[Boolean] and
+      (__ \ "returnFlag").readNullable[Boolean]
+        .orElse((__ \ "returnFlag").readNullable[String].flatMap[Option[Boolean]] {
```

Comment:
can this not be simplified to 
case Some("Y") =>
case Some("N") =>
case Some(x)     =>
case None          =>